### ISSUE TITLE
Exposing a Dart method to JS and calling it in with the wrong number of arguments differs in behavior between dart2JS and DDC

### ISSUE THREAD

**Issue by jimyi** [dart-lang/sdk#31964]
# Describe your issue compiling with DDC
Exposing a Dart method to JS and calling it in with the wrong number of arguments differs in behavior between dart2JS and DDC

# Does this error show up at build time or at run time?
Runtime - DDC makes the method call whereas dart2js throws an error

# Include repro instructions. How can we try to build (or run) this code?

```
import 'dart:js';

context['dartMethod'] = (firstName, lastName) {
  print('$firstName $lastName');
};
```
In the chrome console, try:
`dartMethod("john");`

DDC output:

> john null

dart2js output:

> Uncaught Error: NoSuchMethodError: method not found: 'call'
> Receiver: Closure 'aNw'
> Arguments: ["john"]

**Comment by vsmenon** [comment]
At a minimum, if you use `allowInterop`, I think DDC should provide the same semantics and checking.

Without `allowInterop`, there is an argument we should make it an error in DDC too.

**Comment by jimyi** [comment]
Just edited my earlier comment - I realize I said the wrong thing.  I meant with allowInterop, DDC and dart2js behave the same as they did without allowInterop

**Comment by vsmenon** [comment]
@jimyi - do you mean that dart2js does not work even with `allowInterop`?

**Comment by jimyi** [comment]
That is correct, dart2js throws the same NoSuchMethodError with allowInterop
